src/push/apns.rs

```diff
@@ -76,17 +76,20 @@ pub async fn send_push(
 
     // Notification payload
     let mut payload = if bundle_id.ends_with(".voip") {
-        // This is a voip push, so use the SilentNotificationBuilder
-        SilentNotificationBuilder::new().build(&push_token.0, options)
+        // This is a voip push, so use notification without body but `content-available` set to 1 to allow device wakeup
+        // even though push is empty
+        DefaultNotificationBuilder::new()
+            .set_content_available()
+            .build(&push_token.0, options)
     } else {
         // Regular push, build notification ourselves for full control
         Payload {
             options,
             device_token: &push_token.0,
             aps: APS {
-                alert: Some(APSAlert::Plain("Threema Web Wakeup")),
+                alert: Some(APSAlert::Body("Threema Web Wakeup")),
```

Comment:
In the documentation, 'Plain' notifications seem to refer to a [notification with a body](https://github.com/WalletConnect/a2/blob/11f3d75883dedbe187d0f0a9930eed5eb0aaafae/src/lib.rs#L31), but I'm not entirely sure

Reply:
I logged the request both for VoIP-pushes and for regular pushes with the current code and the updated code (by editing the hyper sourcecode).

Headers

```
Old Regular: {"content-type": "application/json", "apns-priority": "10", "apns-expiration": "1699522255", "apns-topic": "com.example.app", "authorization": "Bearer XXX", "content-length": "140"}
New Regular: {"content-type": "application/json", "apns-priority": "10", "apns-expiration": "1699522515", "apns-topic": "com.example.app", "authorization": "Bearer XXX", "content-length": "140"}
Old Voip:    {"content-type": "application/json", "apns-priority": "10", "apns-expiration": "1699522299", "apns-topic": "com.example.app.voip", "authorization": "Bearer XXX", "content-length": "95"}
New Voip:    {"content-type": "application/json", "apns-priority": "10", "apns-expiration": "1699522925", "apns-topic": "com.example.app.voip", "authorization": "Bearer XXX", "content-length": "115"}
```

Request body:

```
Old Regular: {"3mw":{"wca":null,"wcs":"123deadbeef","wct":1699522165,"wcv":3},"aps":{"alert":"Threema Web Wakeup","mutable-content":1,"sound":"default"}}
New Regular: {"3mw":{"wca":null,"wcs":"123deadbeef","wct":1699522425,"wcv":3},"aps":{"alert":"Threema Web Wakeup","mutable-content":1,"sound":"default"}}
Old Voip:    {"3mw":{"wca":null,"wcs":"123deadbeef","wct":1699522209,"wcv":3},"aps":{"content-available":1}}
New Voip:    {"3mw":{"wca":null,"wcs":"123deadbeef","wct":1699522835,"wcv":3},"aps":{"content-available":1,"mutable-content":0}}
```

The only difference is the `mutable-content` field, which is now being set for Voip requests in the new code. The [Apple docs](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/generating_a_remote_notification) document the field like this:

> The notification service app extension flag. If the value is 1, the system passes the notification to your notification service app extension before delivery.

Based on this, I think that the two requests are equivalent.